apps/rush-lib/src/cli/RushCommandLineParser.ts

```diff
@@ -71,8 +72,18 @@ export class RushCommandLineParser extends CommandLineParser {
   }
 
   protected onExecute(): Promise<void> {
+    // Defensively set the exit code to 1 so if Rush crashes for whatever reason, we'll have a nonzero exit code.
+    // For example, NodeJS currently has the inexcusable design of terminating with zero exit code when
+    // there is an uncaught promise exception.  This will supposedly be fixed in NodeJS 9.
+    // Ideally we should do this for all the Rush actions, but "rush build" is the most critical one
+    // -- if it falsely appears to succeed, we could merge bad PRs, publish empty packages, etc.
+    process.exitCode = 1;
+
```

Comment:
Don't we already do this somewhere?

Reply:
Ah you moved it from here [https://github.com/Microsoft/web-build-tools/blob/master/apps/rush-lib/src/cli/actions/BaseRushAction.ts](https://github.com/Microsoft/web-build-tools/blob/master/apps/rush-lib/src/cli/actions/BaseRushAction.ts)

---
In reply to: [217270112](https://github.com/Microsoft/web-build-tools/pull/823#discussion_r217270112) [](ancestors = 217270112)